docs/core-protocol/public-lock/minting-keys.md

```diff
@@ -0,0 +1,37 @@
+sidebar_position: 2
+---
+
+In the Unlock Protocol ecosystem, Membership Non Fungible Tokens are called [Keys](https://unlock-protocol.com/guides/locksmith-lingo/). Keys are minted from the Lock contract.
+
+There are two methods that can be used to mint Keys. Keys can either be **purchased** or **granted**. Keys can be minted one-by-one or in batches (multiple Keys minted at one time) using either method.
+
+### Purchasing Keys
+
+Let's start by noting that this flow uses the function [`purchase`](/core-protocol/smart-contracts-api/PublicLock#purchase) from the lock even if the keys are free. The best way to describe this flow is that it is _self-served_, which means that anyone can call this function to mint an NFT, whether these keys are free or paid.
+
+:::info
+Note: To disable purchases completely, the best approach is to set the maximum number of keys available for sale to be `0`.
+:::
+
+The `purchase` function can be used to purchase multiple keys at once, and the function takes arguments which are all `arrays` in order to accommodate this. Each of these arrays needs to be of the same size.
+
+One of the arguments is an array of recipients, which means that the wallet sending the transaction can be _different_ from the one who will receive the NFT memberships. This can be very useful when, for example, the application is buying on behalf of users or when a payment happens "off-chain" (using credit card or other mechanism) and the payment provider then wants to mint the NFT membership to the user.
+
+Another argument is an array of key managers. The [key manager](/core-protocol/public-lock/access-control#keymanager) is the address that has the transfer and cancellation rights over the NFT being minted. For credit card purchases where the transaction cannot be considered final, we strongly advise this key manager to be controlled by the entity that triggered the card payment so that if the transaction is reversed, the NFT membership can also be cancelled.
+
+### Granting Keys
+
+The Lock contract also has a [`grantKeys`](/core-protocol/smart-contracts-api/PublicLock#grantkeys) method. This method can _only_ be called by [lock managers](/core-protocol/public-lock/access-control#lockmanager) or [key granters](/core-protocol/public-lock/access-control#keygranter) and can be used to mint/grant keys _for free_. This function does **not require** a payment and is not limited by the maximum number of keys for sale.
+
+Contrary to the previous method, the granter can also customize the expiration of each membership minted using this mechanism, which makes it a convenient method to grant _previews_ or free trials. However, there again, it is critical to set the [key manager](/core-protocol/public-lock/access-control#keymanager) accordingly in order to avoid abuses. For example, if the contract allows for refunds upon cancellation, a malicious recipient of a key airdroped could then claim a refund if they have also been set as _key manager_.
```

Comment:
```suggestion
Contrary to the previous method, the granter can also customize the expiration of each membership minted using this mechanism, which makes it a convenient method to grant _previews_ or free trials. However, there again, it is critical to set the [key manager](/core-protocol/public-lock/access-control#keymanager) accordingly in order to avoid abuses. For example, if the contract allows for refunds upon cancellation, a malicious recipient of an airdropped key could then claim a refund if they have also been set as _key manager_.
```